vipps-psp-api.md

```diff
@@ -204,6 +204,123 @@ Note 93 is for when the makePayment request from Vipps contains the statuses CAN
+
+![PSP API sequence diagram](diagrams/3DSFallbackFlow.png)
+
+Format of MakePaymentRequest response provided by the PSP in case of a soft decline
+
+```json
+{
+  "errorMessage": {
+    "errorId": 1036,
+    "errorText": "Soft Decline"
+  },
+  "paymentInfo": {
+    "pspTransactionId": "7686f7788898767977",
+    "status": "Challenge",
+    "3DSUrl": "https://terminal.com/example"
+  }
+}
+```
+
+The Vipps App will then host the URL in an iframe, letting the user complete the 3DSecure flow. Vipps will then forward the data from the 3DSecure session. The data will be sent in a an otherwise identical version of the MakePaymentRequest that got the soft decline.
+
+```json
+Authorization: makePaymentToken
+{
+  "pspTransactionId": "7686f7788898767977",
+  "merchantSerialNumber": "123456",
+  "cardData": "f0a29801b4#d4ff30e221fa2980ff30e2",
+  "confirmed": "YES/TIMEOUT/CANCEL",
+  "3DSecureData": {
+    "pares" : "eNpdU8tymzAU3ecrvMumYz1AgD2yZnDsTpMZx06TRdudLK5skhiIgNjust/Tr+qXVMIm4DDDoHvuOdKZexB/2hqA2SOo2oC4Ggz4AspSbmCQJpPrSELAZKApSOyFkWYaY+brJAq0pzTQa6ewmlX8Hd4aRZDoKMIs8WXiK2/NvIAFDPwwoRrCkV6fFVbzDqZM80yQIR5SjtqybRcyE4xFPmUhoZiGhPkRRw5tGQswaiuzqgUsJNXb9PZeMBpYFUfnsuvvwNzOBKa4fTg6QR0lkzsQ84PcFa/AUVN1TZXXWWWOgmLrpS26dm1exbaqijFC+/1+CKddhrnZII5cs7WOPnvnq9oBZf+wQ5qIxSzed+/8uPj9QO6flb98iiccOUbHT2QF1hlhmODRgARj6o0Z46jBezPaOd/i35+/hH7x7ATOQMconJf4hBLqKH2kN43aGMjUUYxCN4626ghwKPIMrMbm+7HuGYZSCevPfT4m83kQ/ObbRcCqsnHls7V++fVAHoPlnb/9eVMuVz+m8fzrNN5MXOwN6cJHaoMiETkZSbvUOGr3t0e7v7i5A+h8CcQVR5cX5D/FxN2u"
+  }
+}
+```
+
+# Recurring PSP payments
+
+The PSP API supports recurring payments out of the box. This enables the PSP to perform recurring payments through Vipps, while retaining full transactional control. This has been built as an extension to the existing PSPv2 API, and no existing integrations will be affected, other than the possibility to initialize and preform recurring payments.
+
+## Subscription and oneclick scopes
+
+As of now, there are two possible ways to perform a recurring payment - *subscription* and *oneclick*. These are referred to as the *scope* of the recurring agreement. Only one *scope* can be used at a time, and it's not possible to change the scope of an agreement.
+
+*Subscription* based payments are created as a consent to an agreement that allows the PSP to withdraw money on unknown intervals. This implies that the user won't have to accept the payment on each occasion, only the first one when consenting to the agreement. An example of this could be a subscription to a music streaming service.
+
+*Oneclick* payments are created as agreements where the user will be prompted on each payment. An example of a oneclick payment might be a ticketing service where the user will get prompted to buy a new ticket when the old one expires. Instead of switching to the app to accept the payment, the user already has a oneclick recurring agreement with Vipps as the payment source, and the ticketing app won´t need to involve the Vipps app at all.
```

Comment:
Do we still have oneclick in this form?

Reply:
Good catch, we are not launching with oneclick